scylla/src/transport/errors.rs

```diff
@@ -870,6 +870,38 @@ pub enum CqlEventHandlingError {
     SendError,
 }
 
+/// Failed to execute a retriable user request.
+#[derive(Error, Debug, Clone)]
+#[non_exhaustive]
+pub enum RetriableRequestError {
```

Comment:
I also dislike this name. I just couldn't come up with anything better. My only idea is to rename `UserRequestError` -> `[User]RequestAttemptError` (I like this name, because it suggests that it's an error of a single attempt). Then we could use `UserRequestError` name instead of `RetriableRequestError`.

What do you think of `TimeoutableRequestError` name (introduced in this PR)? I'm not sure if you reached the commit that introduces it - this enum simply represents either a request timeout, or a request failure (current `RetriableRequestError`). I wonder if there is a better name for this one as well.

Reply:
> My only idea is to rename `UserRequestError` -> `[User]RequestAttemptError` (I like this name, because it suggests that it's an error of a single attempt). Then we could use `UserRequestError` name instead of `RetriableRequestError`.

Sounds good, let's do that.

> What do you think of `TimeoutableRequestError` name (introduced in this PR)? I'm not sure if you reached the commit that introduces it - this enum simply represents either a request timeout, or a request failure (current `RetriableRequestError`). I wonder if there is a better name for this one as well.

`MaybeTimeoutedRequestError` is one idea that I've had, but it's not ideal, too.